test/test_nn.py

```diff
@@ -3529,6 +3529,27 @@ def test_adaptive_pooling_avg_nhwc(self):
         self.assertEqual(out, ref_out)
         self.assertEqual(input.grad, ref_input.grad)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_adaptive_pooling_avg_nhwc_propagation(self):
+        input = torch.randint(1, 10, (4, 8, 1, 1), dtype=torch.float32, device="cuda")
+        input = input.contiguous(memory_format=torch.channels_last).requires_grad_()
+        grad = torch.randint(1, 10, (4, 8, 4, 4), dtype=torch.float32, device="cuda")
+        pool = torch.nn.AdaptiveAvgPool2d((4, 4))
+
+        ref_input = input.detach().clone().contiguous().requires_grad_(True)
+        ref_grad = grad.detach().clone().contiguous()
+        ref_pool = torch.nn.AdaptiveAvgPool2d((4, 4)).cuda()
+
+        out = pool(input)
+        out.backward(grad)
```

Comment:
How does backward work with contiguous (not channels-last) grad?

Reply:
I will add a test here.

Kernel would output grad_input as the same memory_format with input, handles grad_output as it (so memory accessing pattern will be bad, but my intuition is, it should still be better than another permutation kernel, I should have benchmarked it :smile: )